instrumentation/net/http/otelhttp/test/handler_test.go

```diff
@@ -50,8 +50,8 @@ func assertScopeMetrics(t *testing.T, sm metricdata.ScopeMetrics, attrs attribut
 	require.Len(t, sm.Metrics, 3)
 
 	want := metricdata.Metrics{
-		Name:        "http.server.request_content_length",
-		Description: "Measures the size of HTTP request content length (uncompressed)",
+		Name:        "http.server.request.size",
```

Comment:
This looks to me like a breaking change. Why wasn't it communicated as such?

Reply:
@SharpTech sorry about that.  I've opened https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4898 to add it to the changelog.

Reply:
Hi @dashpole wouldn't it be better to rollback so the deprecation statement remain true?

Reply:
The deprecation statements refer to these constants: https://github.com/open-telemetry/opentelemetry-go-contrib/pull/4707/files#diff-be9e2412d1919396470e196d2be4bddf6997059892167fd8381a9cfa478933faR34

> I also think breaking stuff with a feature release isn't the right approach.

The otelhttp module is not stable, and is still released under a v0.* version, so breaking changes are expected.  You can see the versioning policy here: https://github.com/open-telemetry/opentelemetry-go/blob/main/VERSIONING.md.  The telemetry in otelhttp follows the semantic conventions for http metrics: https://opentelemetry.io/docs/specs/semconv/http/http-metrics/, so breaking changes to conventions require us to make breaking changes here.

Reply:
I understand, but users like me running this in production. We need to deprecate fields before we can upgrade. If I would want to use the new metrics I couldn't. I'm saying the feature and the breaking change could have been shipped in two PR  (releases).